File: 9020번： 골드바흐의 추측/골드바흐의 추측.c
/* ************************************************************************** */
/*                                                                            */
/*                                                      :::    :::    :::     */
/*   Problem Number: 9020                              :+:    :+:      :+:    */
/*                                                    +:+    +:+        +:+   */
/*   By: luchael <boj.kr/u/luchael>                  +#+    +#+          +#+  */
/*                                                  +#+      +#+        +#+   */
/*   https://boj.kr/9020                           #+#        #+#      #+#    */
/*   Solved: 2025/02/25 14:55:43 by luchael       ###          ###   ##.kr    */
/*                                                                            */
/* ************************************************************************** */
#include <stdio.h>
int main()
{
    int p = 2, cnt = 0, n, T, prime[10001];
    for (int i = 0; i < 10001; i++)
        prime[i] = 1;
    while (p * p <= 10000)
    {
        if (prime[p] == 1)
            for (int i = p * 2; i < 10001; i += p)
                prime[i] = 0;
        p++;
    }
    prime[0] = 0, prime[1] = 0;
    scanf("%d", &T);
    for (; T > 0; T--)
    {
        scanf("%d", &n);
        int up = n / 2, down = n / 2;
        for (; down > 1; down--, up++)
        {
            if (prime[down] && prime[up])
            {
                printf("%d %d\n", down, up);
                break;
            }
        }
    }
}

File: 6588번： 골드바흐의 추측/골드바흐의 추측.py
#  **************************************************************************  #
#                                                                              #
#                                                       :::    :::    :::      #
#    Problem Number: 6588                              :+:    :+:      :+:     #
#                                                     +:+    +:+        +:+    #
#    By: luchael <boj.kr/u/luchael>                  +#+    +#+          +#+   #
#                                                   +#+      +#+        +#+    #
#    https://boj.kr/6588                           #+#        #+#      #+#     #
#    Solved: 2025/02/25 14:50:09 by luchael       ###          ###   ##.kr     #
#                                                                              #
#  **************************************************************************  #
import sys

input = sys.stdin.readline

prime = [True]*1000001

for i in range(3,1001,2):

    if prime[i]:

        for j in range(i,1000000//i+1,2):

            prime[i*j]=False

while True:

    num=int(input())

    if num==0: break

    a,b=3,num-3

    while b>=a:

        if(prime[a] and prime[b]):

            print(num,'=',a,'+',b)

            break

        a+=2

        b-=2

